src/SFA.DAS.LevyTransferMatching/Data/Models/Pledge.cs

```
@@ -91,6 +91,11 @@ public void Credit(int creditAmount, UserInfo userInfo)
             RemainingAmount += creditAmount;
         }
 
+        public void ClosePledge()
+        {
```

I'd expect a check here to ensure that the pledge was active - if it's closed already, for example, we could throw an invalidoperation exception to indicate this. It is surely an error to close a closed pledge.

There should be change tracking here, something like this:
```
            StartTrackingSession(UserAction.ClosePledge, userInfo);
            ChangeTrackingSession.TrackUpdate(this);
```
(You'll need to add the `UserAction` enum value).

---

This check has been implemented around the Pledge Status. If the Pledge is already closed, this will throw an InvalidOperationException. Also, the tracking session has been added here with a new UserAction enum.